src/arch/ucx/machine-onesided.C

```diff
@@ -147,21 +155,38 @@ void LrtsIssueRput(NcpyOperationInfo *ncpyOpInfo)
     UCX_LOG(4, "srcPE %d destPE %d, size %d",
             ncpyOpInfo->srcPe, ncpyOpInfo->destPe, ncpyOpInfo->ncpyOpInfoSize);
 
-    if (ncpyOpInfo->isDestRegistered != 0) {
-        UcxRmaOp(ncpyOpInfo, UCX_RMA_OP_PUT);
-    } else {
-        // Remote buffer is not registered, ask peer to perform get
-        UcxSendMsg(CmiNodeOf(ncpyOpInfo->destPe), ncpyOpInfo->destPe,
-                   ncpyOpInfo->ncpyOpInfoSize, (char*)ncpyOpInfo,
-                   UCX_RMA_TAG_GET, UcxRmaSendCompleted);
-        UCX_LOG(4, "Sending Get REQ to %d, mem size %d", ncpyOpInfo->destPe,ncpyOpInfo->srcSize);
-
-    }
+    // Since UCX doesn't guarantee destination completion with PUT, it is implemented
+    // as a reverse operation with the remote destination performing a GET
```

Comment:
maybe it is worth to register src here (if needed)? Otherwise remote side will have to request registration

Reply:
Since `LrtsIssueRput` is invoked on the source, the higher layer code (ck-core) ensures that the local (source) buffer is always registered if it is required. (at /ck-core/ckrdma.C:190).